### PR TITLE
Fix LTTng-tools builds for versions prior to 2.8 when disabling UST

### PR DESCRIPTION
**Pull request by frdeso**

### CODE REVIEW

File: vlttng/venv.py [hunk 1]
```diff
@@ -554,7 +554,11 @@ def _build_lttng_tools(self):
         if 'lttng-ust' in self._profile.projects:
             add_args = '--with-lttng-ust-prefix={}'.format(shlex.quote(self._paths.usr))
         else:
+            # LTTng-Tools prior to v2.8 was using a different flag to turn off
+            # UST support. We add both to the configure script arguments since the
+            # unsupported one will be ignored.
             add_args = '--without-lttng-ust'
+            add_args += ' --disable-lttng-ust'
```
Comment (eepp) [inline vlttng/venv.py]: Why not:

```
add_args = '--without-lttng-ust --disable-lttng-ust'
```

Also, the `--disable-lttng-ust` option (and `--enable-lttng-ust=no` I guess) should be detected in line 550 above for the warning message. At this point I would add all those strings to a list and for each of them check if its in `project.configure` (and generate that part of the warning message with `'/'.join(the_list)`).

**Review by eepp — CHANGES_REQUESTED** [review]
Almost there!

**Comment by frdeso** [comment]
see new version of the patch

File: vlttng/venv.py [hunk 2]
```diff
@@ -547,14 +547,18 @@ def _build_lttng_tools(self):
         if project is None:
             return
 
-        if '--with-lttng-ust-prefix' in project.configure or '--without-lttng-ust' in project.configure:
-            msg = 'I would not pass the --with-lttng-ust-prefix/--without-lttng-ust configure option if I were you: they are handled by vlttng'
-            _pwarn(msg)
+        for ust_config in ['--with-lttng-ust-prefix', '--without-lttng-ust', '--disable-lttng-ust']:
+            if ust_config in project.configure:
+                msg = 'I would not pass the '+ust_config+' configure option if I were you: they are handled by vlttng'
```
Comment (eepp) [inline vlttng/venv.py]:
```
'/'.join(ust_config)
```

File: vlttng/venv.py [hunk 3]
```diff
@@ -547,14 +547,18 @@ def _build_lttng_tools(self):
         if project is None:
             return
 
-        if '--with-lttng-ust-prefix' in project.configure or '--without-lttng-ust' in project.configure:
-            msg = 'I would not pass the --with-lttng-ust-prefix/--without-lttng-ust configure option if I were you: they are handled by vlttng'
-            _pwarn(msg)
+        for ust_config in ['--with-lttng-ust-prefix', '--without-lttng-ust', '--disable-lttng-ust']:
```
Comment (eepp) [inline vlttng/venv.py]: Name this list, add `--enable-lttng-ust=no`.

File: vlttng/venv.py [hunk 4]
```diff
@@ -547,14 +547,18 @@ def _build_lttng_tools(self):
         if project is None:
             return
 
-        if '--with-lttng-ust-prefix' in project.configure or '--without-lttng-ust' in project.configure:
```
Comment (eepp) [inline vlttng/venv.py]: Use the named list here:

```
for opt in lttng_ust_opts:
    if opt in project.configure:
        # ...
```

**Review by eepp — CHANGES_REQUESTED** [review]

File: vlttng/venv.py (hunk 3 above)
Comment (frdeso) [reply vlttng/venv.py]: Should I include `--enable-lttng-ust=yes` as well?

File: vlttng/venv.py (hunk 2 above)
Comment (frdeso) [reply vlttng/venv.py]: Do you mean `'/'.join(lttng_ust_opts)`?

File: vlttng/venv.py (hunk 3 above)
Comment (eepp) [reply vlttng/venv.py]: Yes, in fact, just include `--enable-lttng-ust`.

File: vlttng/venv.py (hunk 2 above)
Comment (eepp) [reply vlttng/venv.py]: Yes.

Comment (eepp) [reply vlttng/venv.py]: You will need to rebase this BTW, there has been some changes in this file.

Comment (frdeso) [reply vlttng/venv.py]: using `join()` the entire list will be printed for each `lttng_ust_ops` present in the configure options.
If I have both `--with-lttng-ust-prefix`  and `--enable-lttng-ust=yes`, the same warning will be printed twice.
If that's what we want, we should break after the first warning.

Comment (eepp) [reply vlttng/venv.py]: Yes you're right, no join then.

Also, prefer using a tuple than a list for `lttng_ust_opts`.

**Comment by frdeso** [comment]
see new version of the patch

**Pull request closed without merging**